bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/DefaultModelValidator.java

```diff
@@ -0,0 +1,136 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
+ * available at https://opensource.org/licenses/MIT.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
+ ********************************************************************************/
+package org.eclipse.emfcloud.modelserver.emf.common;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.emf.common.util.BasicDiagnostic;
+import org.eclipse.emf.ecore.EAttribute;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EClassifier;
+import org.eclipse.emf.ecore.EDataType;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.util.Diagnostician;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.emf.ecore.util.ExtendedMetaData;
+import org.eclipse.emfcloud.modelserver.emf.configuration.FacetConfig;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+
+public class DefaultModelValidator implements ModelValidator {
+
+   protected final ModelRepository modelRepository;
+   protected final FacetConfig facetConfig;
+
+   @Inject
+   private ObjectMapper mapper;
+
+   @Inject
+   public DefaultModelValidator(final ModelRepository modelRepository, final FacetConfig facetConfig) {
+      this.modelRepository = modelRepository;
+      this.facetConfig = facetConfig;
+   }
+
+   @Override
+   public JsonNode validate(final String modeluri, final ObjectMapper mapper) {
+      Optional<EObject> model = this.modelRepository.getModel(modeluri);
+      Optional<Resource> res = this.modelRepository.loadResource(modeluri);
+      if (res.isPresent()) {
```

Comment:
Ah sorry, my bad! I oversaw that you use the `model` variable below... Could you just combine the checks in this if clause so we are on the safe side!
```suggestion
      if (model.isPresent() && res.isPresent()) {
```